index.js

```diff
@@ -1,63 +1,271 @@
+  if (!(angle > 180 && shortcut)) {
+    while (step) {
+      bearing = bearing + Math.pow(-1, !right + 1) * resMultiple;
+      spoke = destination(pt, radius, bearing, units);
+      arc.push(spoke.geometry.coordinates);
+      step--;
     }
-    buffered = featurecollection([buffered])
   }
-  else{
-    buffered = featurecollection([polygon(buffered.coordinates)])
+  // Add spoke for bearing 2, but only if this spoke has not been added yet. Do this by checking the destination point, since slightly different bearings can create equal destination points.
+  var spokeBearing2 = destination(pt, radius, bearing2, units);
+  if (!spokeBearing2.geometry.coordinates.equals(spoke.geometry.coordinates)) {
+    arc.push(spokeBearing2.geometry.coordinates);
   }
+  return helpers.lineString(arc)
+}
 
-  return buffered;
-}
\ No newline at end of file
+function filterNetWinding(fc, filterFn) {
+  var i = fc.features.length;
+  while (i--) {
+    if (!filterFn(fc.features[i].properties.netWinding)) {
+        fc.features.splice(i, 1);
+    } else {
+      fc.features[i].properties = {};
+    }
+  }
+  return fc;
+}
+
+function unionFeatureCollection(fc) {
+  // Note: union takes a polygon, but return a polygon or multipolygon (which it can not take in). In case of buffes, however, it will always return a polygon
+  if (fc.features.length == 0) return {type: "Feature", geometry: null};
+  var incrementalUnion = fc.features[0];
+  if (fc.features.length == 1) return incrementalUnion;
+  for (var i = 1; i < fc.features.length; i++) {
+    incrementalUnion = union(incrementalUnion, fc.features[i]);
+  }
+  return incrementalUnion
+}
+
+function offsetToBuffer(polygonOffset) {
+  var unionWithWindingOne = unionFeatureCollection(filterNetWinding(simplepolygon(polygonOffset), function (netWinding){return netWinding == 1}));
+  var unionWithWindingZero = unionFeatureCollection(filterNetWinding(simplepolygon(polygonOffset), function (netWinding){return netWinding == 0}));
+  // This last one might have winding -1, so we might have to rewind it if the difference algorithm requires so
+
+  if (unionWithWindingOne.geometry == null) return {type: "Feature", geometry: null};
+  if (unionWithWindingZero.geometry == null) return unionWithWindingOne;
+  return difference(unionWithWindingOne, unionWithWindingZero);
+}
+
+// This function awaits possible future use
+function winding(poly){
+  // compute winding of first ring
+  var coords = poly.geometry.coordinates[0];
+  var leftVtx = 0;
+  for (var i = 0; i < coords.length-1; i++) { if (coords[i][0] < coords[leftVtx][0]) leftVtx = i; }
+  return (coords[(leftVtx-1).mod(coords.length-1)][1] > coords[(leftVtx+1).mod(coords.length-1)][1]) ? 1 : -1;
+}
+
+// This function awaits possible future use
```

Comment:
It looks like this method & the winding method are both currently used - are these comments inaccurate?